iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -1085,14 +1120,58 @@ static MQTT_DEVICE_TWIN_ITEM* createDeviceTwinMsg(MQTTTRANSPORT_HANDLE_DATA* tra
 static int publishDeviceTwinGetMsg(MQTTTRANSPORT_HANDLE_DATA* transport_data, MQTT_DEVICE_TWIN_ITEM* mqtt_info)
 {
     int result;
+    STRING_HANDLE msg_topic = NULL;
+
+    // Construct the topic
+    if (mqtt_info->device_twin_msg_type == RETRIEVE_PROPERTIES_FULL)
+    {
+        msg_topic = STRING_construct_sprintf(GET_PROPERTIES_TOPIC, mqtt_info->packet_id);
+    }
+    else if(mqtt_info->device_twin_msg_type == RETRIEVE_PROPERTIES_DESIRED)
+    {
+        msg_topic = STRING_construct_sprintf(GET_DESIRED_PROPERTIES_TOPIC, mqtt_info->packet_id);
+    }
+    else if(mqtt_info->device_twin_msg_type == RETRIEVE_PROPERTIES_REPORTED)
+    {
+        msg_topic = STRING_construct_sprintf(GET_REPORTED_PROPERTIES_TOPIC, mqtt_info->packet_id);
+    }
 
-    STRING_HANDLE msg_topic = STRING_construct_sprintf(GET_PROPERTIES_TOPIC, mqtt_info->packet_id);
     if (msg_topic == NULL)
     {
         LogError("Failed constructing get Prop topic.");
         result = MU_FAILURE;
     }
     else
+    {
+        //Append if-not-version property
+        if (mqtt_info->twinRequestOptions)
+        {
+            int64_t device_current_twin_version;
+
+            if (mqtt_info->twinRequestOptions->is_current_version_set)
+            {
+                if (STRING_sprintf(msg_topic, "%s%s=%"PRIi64, PROPERTY_SEPARATOR, IF_NOT_VERSION_PROPERTY, mqtt_info->twinRequestOptions->current_version) != 0)
+                {
+                    LogError("Failed constructing get Prop topic.");
+                    result = MU_FAILURE;
+                }
+                else
+                {
+                    result = 0;
+                }
+            }
+            else
+            {
+                result = 0;
+            }
+        }
+        else
+        {
+            result = 0;
+        }
+    }
+
+    if (result != MU_FAILURE)
```

Comment:
I could have sworn `MU_FAILURE` expands to `__LINE__` so I think this would be different than the one on 1156. Consider: 
```suggestion
    if (result == 0)
```

Reply:
I see.  It can be either 1 or __LINE__.  thanks for this catch.